eth-contracts/contracts/ServiceTypeManager.sol

```diff
@@ -109,46 +111,23 @@ contract ServiceTypeManager is InitializableV2 {
         uint lastIndex = validServiceTypes.length - 1;
         validServiceTypes[serviceIndex] = validServiceTypes[lastIndex];
         validServiceTypes.length--;
-        // Overwrite values
-        serviceTypeStakeRequirements[_serviceType].minStake = 0;
-        serviceTypeStakeRequirements[_serviceType].maxStake = 0;
-    }
-
-    /**
-     * @notice Update a service type
-     * @param _serviceType - type of service
-     * @param _serviceTypeMin - minimum stake for service type
-     * @param _serviceTypeMax - maximum stake for service type
-     */
-    function updateServiceType(
-        bytes32 _serviceType,
-        uint _serviceTypeMin,
-        uint _serviceTypeMax
-    ) external
-    {
-        _requireIsInitialized();
-        require(
-            msg.sender == governanceAddress,
-            "Only callable by Governance contract"
-        );
-
-        require(this.serviceTypeIsValid(_serviceType), "Invalid service type");
 
-        serviceTypeStakeRequirements[_serviceType].minStake = _serviceTypeMin;
-        serviceTypeStakeRequirements[_serviceType].maxStake = _serviceTypeMax;
+        // Mark as invalid
+        serviceTypeInfo[_serviceType].isValid = false;
```

Comment:
i think this is fine but want to confirm - can we re-register a previously known serviceType after removing it with different min/max bounds?

Reply:
good catch - it has been possible to re-add a previously removed serviceType with diff bounds, which would expose the same accounting vulnerability. This has been locked down